src/main/java/graphql/schema/Coercing.java

```diff
@@ -1,7 +1,7 @@
 package graphql.schema;
 
 
-public interface Coercing {
+public interface Coercing <T2> {
```

Comment:
I would highly suggest to use T (or K,V ..) instead T2. T2 is a very unusual Type Parameter name.